src/sidebar/components/test/hypothesis-app-test.js

```diff
@@ -325,75 +386,113 @@ describe('hypothesisApp', function () {
+      it('discards draft annotations', function () {
+        var ctrl = createController();
 
-      ctrl.logout();
+        ctrl.logout();
 
-      assert(fakeDrafts.discard.calledOnce);
-    });
+        assert(fakeDrafts.discard.calledOnce);
+      });
 
-    it('does not emit "annotationDeleted" if the user cancels the prompt', function () {
-      var ctrl = createController();
-      fakeDrafts.count.returns(1);
-      $rootScope.$emit = sandbox.stub();
-      fakeWindow.confirm.returns(false);
+      it('does not emit "annotationDeleted" if the user cancels the prompt', function () {
+        var ctrl = createController();
+        fakeDrafts.count.returns(1);
+        $rootScope.$emit = sandbox.stub();
+        fakeWindow.confirm.returns(false);
 
-      ctrl.logout();
+        ctrl.logout();
 
-      assert($rootScope.$emit.notCalled);
-    });
+        assert($rootScope.$emit.notCalled);
+      });
 
-    it('does not discard drafts if the user cancels the prompt', function () {
-      var ctrl = createController();
-      fakeDrafts.count.returns(1);
-      fakeWindow.confirm.returns(false);
+      it('does not discard drafts if the user cancels the prompt', function () {
+        var ctrl = createController();
+        fakeDrafts.count.returns(1);
+        fakeWindow.confirm.returns(false);
+
+        ctrl.logout();
+
+        assert(fakeDrafts.discard.notCalled);
+      });
+
+      it('does not prompt if there are no drafts', function () {
+        var ctrl = createController();
+        fakeDrafts.count.returns(0);
 
-      ctrl.logout();
+        ctrl.logout();
 
-      assert(fakeDrafts.discard.notCalled);
+        assert.equal(fakeWindow.confirm.callCount, 0);
+      });
+    }
+
+    context('when no third-party service is in use', function () {
+      doSharedTests();
+
+      it('calls session.logout()', function () {
+        var ctrl = createController();
+        ctrl.logout();
+        assert.called(fakeSession.logout);
+      });
     });
 
-    it('does not prompt if there are no drafts', function () {
-      var ctrl = createController();
-      fakeDrafts.count.returns(0);
+    context('when a third-party service is in use', function () {
+      beforeEach('configure a third-party service to be in use', function() {
+        fakeServiceConfig.returns({});
+      });
+
+      doSharedTests();
 
-      ctrl.logout();
+      it('sends LOGOUT_REQUESTED if a third-party service is in use', function () {
+        createController().logout();
 
-      assert.equal(fakeWindow.confirm.callCount, 0);
+        assert.calledOnce(fakeBridge.call);
+        assert.calledWithExactly(fakeBridge.call, bridgeEvents.LOGOUT_REQUESTED);
+      });
+
+      it('does not send LOGIN_REQUESTED if the user cancels the prompt', function () {
+        fakeDrafts.count.returns(1);
+        fakeWindow.confirm.returns(false);
+
+        createController().logout();
+
+        assert.notCalled(fakeDrafts.discard);
```

Comment:
This assert doesn't verify that the `LOGIN_REQUESTED` event was not sent as specified in the description.